aws_config.go

```diff
@@ -6,18 +6,29 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
+	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
 	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 	"github.com/aws/aws-sdk-go-v2/service/sts"
 	"github.com/aws/aws-sdk-go-v2/service/sts/types"
 	"github.com/aws/smithy-go/logging"
+	"github.com/aws/smithy-go/middleware"
 	"github.com/hashicorp/go-cleanhttp"
 )
 
+const (
+	// appendUserAgentEnvVar is a conventionally used environment variable
+	// containing additional HTTP User-Agent information.
+	// If present and its value is non-empty, it is directly appended to the
```

Comment:
Nit - The value is actually prepended?

```suggestion
	// If present and its value is non-empty, it is directly prepended to the
```

Reply:
The content of `TF_APPEND_USER_AGENT` is appended, while the elements in `Config.UserAgentProducts` are prepended.